tirzepatide sensitivity testing

a-chacko opened the pull request:
In this pull request, changes were made to the following:

1. New sensitivity testing script created - `sensitivity_testing_policy_38.R` and is located - ahl_obesity_blueprint/analysis/policy_analysis/new_policies
2. Additional inputs (Constants) required for benefit estimation added to `config.R`
3. Existing functions used in Tirzepatide modelling script `policy_38.R` was moved to `utils.R` for easy calling in sensitivity analysis scripts.
4. New function to estimate benefits by BMI class added to `post_processing.R` . This function was used in the sensitivity testing to generate benefit estimates for each iteration

All the relevant source files are attached to the script and should run without any dependency issues

a-chacko committed:
tirzepatide sensitivity testing

a-chacko committed:
sensitivity testing files and updated benefit calculation based on agreed method

moniquetan commented:
For this policy (GLP-1s / Tirzepatide), I first list the steps I expected to see (from reading the methods), then check whether, and how, each of them were implemented in the code. 

To help prioritisation, I label each of my comment as MUST vs SHOULD vs COULD, which generally mean the following:
- MUST are comments that need to be addressed before this work is QA’d by Clare
- SHOULD and COULDs are comments that can be left for repository redesign work 

**I bolded the comments that would need addressing today.**

### Expected steps:
**Preprocess health survey datasets**
	1	Subset for variables of interest
	2	Drop rows with missing body weight / height values
	3	Split adults and children
	4	Create the Basal Metabolic Rate (BMR) variable: using the Mifflin St Jeor equation (for adults) and the Henry equation (for children)
	5	Create the Physical Activity Level (PAL) variable: 1.6 for adults, 1.4-1.75 for children (depending on age group)
	6	Create the Total Energy Expenditure (TEE) variable: TEE = BMR * PAL

**Model impact when effect size is expressed as weight unit**
	7	Identify eligible individuals, based on (i) BMI, (ii) number of 'qualifying' comorbidities, and (iii) ethnic group. 'Qualifying' comorbidities consist of: atherosclerotic cardiovascular disease (ASCVD), hypertension, dyslipidaemia, obstructive sleep apnea (OSA), and type 2 diabetes mellitus (T2D).
	8	Assign eligible individuals to cohorts
	9	In each cohort, randomly select individuals until reaching the number of patients specified in the yearly NICE allocations
	10	For each population subgroup, apply change in calories in body weight (depending on T2D status)
	11	No weight regain
	12	Calculate individual participant's new BMI on 365th day of each year
	13	Compute % of individual in each BMI category

**Post-processing (impact)**
 14. Calculate the difference in % of obese individuals at baseline and in 5th year 
 15. Convert difference to relative reduction in obesity prevalence
 
**Post-processing (cost saving)**
16. Calculate cost saving to government by scaling down Frontier Economics estimates (annual cost of obesity = £74 billion, so if obesity is reduced by 10%, value to UK government is £7.4 billion)

**Sensitivity testing** 
17. Run a 1000 iterations of Steps 9-17 and pick the mean values for impact and benefit.

### Checking the code implementation of each step:

**General** 
- I was able to run `analysis/policy_analysis/new_policies/policy_38.R` almost the entire script without errors, up until line 663 (writing `table_outputs`). But I was able to view `table_outputs`, just not to save it, so that’s not an immediate issue. [COULD]
- `NUMBER_OF_PEOPLE_PER_YEAR = 1200000` should be removed? We aren’t treated 1.2 million people each year, in this policy? [SHOULD]

**Steps 1-6 (preprocess health survey datasets)** are done with the custom-made function process_clean_save(), defined in the file `pre_processing/pre_processing_adult.R`. The function is clearly documented and systematically preprocess each variable of interest. For each categorical (factor) variable in both surveys, I assume that the level used is correct. 

**Step 7 (determine intervention eligibility)** is done using the variables available in HSE to identify people with qualifying comorbidities. This isn’t perfect but is making the best possible use of the variables available in HSE. 
- **Could we double-check in HSE’s data dictionary that `diabetes_type == 1` checks for type 2 diabetes diagnosis, and not type 1? [MUST]** This is an important check as this condition is also used to assign weight loss effect size later in the script.
- **A clarifying question: as part of this September 2025 update, are we producing a Tirzepatide number for Scotland? I couldn’t find an answer here** https://docs.google.com/document/d/1NZZMIG8iphFNbARqXtL25Nx7Ou5ZJujb3EPhIck8Vzo/edit?tab=t.9wwraqy3x1cn

**Step 8 (cohort assignment)** is done in a way that matches the guidance (https://www.nice.org.uk/guidance/ta1026/documents/supporting-documentation-7), including adjusted BMI thresholds for non-White ethnic groups. This was converted to yearly numbers in Table A here https://docs.google.com/document/d/1AjMZWK9XV1lOqAGUcSZPaBYAKXHtOPfUz4zJRICpqFc/edit?tab=t.0 and this yearly allocation is reflected correctly in the code. 

**Step 9 (random selection)** includes a workaround for cases where we run out of people to assign, whereby people are taken from previous cohorts (first), and the next cohort (second). While this doesn’t match the guidance, it is a pragmatic way to get enough people treated in the model.

**Step 10 (apply effect size)** uses `assign_weight_changes()` function defined in `analysis/policy_analysis/new_policies/policy_38.R`, which (correctly) assigns weight loss conditional on T2D status. All weight loss is modelled to happen in the first year that people enter the intervention (as opposed to the first year of the model), which is correct. The assumption that all weight loss happens in a year is explicitly stated in the comments, which is helpful, and is consistent with what trial data shows (https://docs.google.com/document/d/1AjMZWK9XV1lOqAGUcSZPaBYAKXHtOPfUz4zJRICpqFc/edit?tab=t.0).

**Step 11 (weight regain**) in the case of Tirzepatide is (correctly) 0. There is redundancy due to legacy GLP-1 modelling accounting for weight regain, but I would agree with not deleting these lines for now.

**Step 12 (estimate post-intervention BMI)** 
- Should amend the code to match NICE’s definition for obesity classes (https://www.nice.org.uk/guidance/ng246/chapter/Identifying-and-assessing-overweight-obesity-and-central-adiposity), i.e. adding an extra class, and renaming the classes [SHOULD]
- NICE guidelines defines adjusted BMI thresholds for obesity class definition for non-White ethnic groups. This ideally should be reflected in our modelling, but this change would affect all policies. A change for PL to discuss with Sponsor before the code gets changed.

**Step 13 (calculate % individuals in each BMI category)**: OK

**Step 14-15 (calculate prevalence reduction)**: done with the custom-made function extract_relative_change(), defined in the file post_processing.R. 
- There is great media interest around Tirzepatide and the effect it would have. The comms team are asking for numbers. As the bulk of the effect would be located at BMIs above 35 and 40, could we amend this function (could be done locally, in this script) to get numbers not just for those moving to a BMI below 30, but also to be a BMI below 35 (ie movement from Class 2 to Class 1) and BMI below 50 (ie movement from Class 3 to Class 2)? You must have done this already to get the benefit figures to Hugo but I wasn’t able to reproduce it with this function. [MUST, for next week (not needed for Clare’s QA, but comms would need the number next week)]

**Step 16 (calculate benefits)**: simple proportional scaling down of the £74 bn cost estimate based on the relative % reduction in obesity prevalence. This is done with the custom-made function extract_pound_benefit(), also defined in the file post_processing.R. Function was amended to attribute a different value to to Class 3 obesity to present option to Sponsor (https://docs.google.com/presentation/d/1vW_9kjWoYoe_xvqEdC6rq15G9TT9F_fVnDik3OEQf98/edit?slide=id.g371f70717f1_0_2651#slide=id.g371f70717f1_0_2651) but these amends are not used. No need to change it again.

**Step 17 (sensitivity testing)** is implemented in `analysis/policy_analysis/new_policies/sensitivity_testing_policy_38.R`, which correctly iterates Steps 9-16 1000 times. 
- Should find a way to iterate the Tirzepatide model without copy-pasting large chunks of code to avoid any mistakes (e.g., making changes in one script that don’t get carried over to the second) [SHOULD]
- **I run 5000 and 10000 iterations and obtained an impact figure of 0.32 (instead of 0.31) and benefit figure of 1.13 (instead of 1.12). Should use 5000 iterations instead of 1000, and use the results for 5000 iterations [MUST].**

a-chacko committed:
sensititvity tests with 5000 iterations

a-chacko commented:
Hi Monique, thanks so much for such a detailed and clear code review.

Collating the different MUST, SHOULD comments:

**MUST**:

1. Could we double-check in HSE’s data dictionary that `diabetes_type == 1` checks for type 2 diabetes diagnosis, and not type 1?
Yes, I have rechecked this and  `diabetes_type == 1` indicates that an individual is living with type 2 diabetes (T2D). The original HSE variable is TypeD and has `TypeD == 2` coded as individual living with T2D. However, in the data preprocessing scripts, I have renamed this variable as `diabetes_type` and recoded all instances where individual has T2D as `1` and all others `0`.

2. A clarifying question: as part of this September 2025 update, are we producing a Tirzepatide number for Scotland? 
At this point, I think, we don't have any Scotland specific roll out plan via GPs using which we could model. The NICE guidelines published last December were specific to England. Maybe something to check later in detail. From some quick searches it appears that the Tirz. for managing excess weight is primarily prescribed via specialist weight management clinics like Semaglutide in England. Probably something for us to keep an eye out on, if something comes up.

3. I run 5000 and 10000 iterations and obtained an impact figure of 0.32 (instead of 0.31) and benefit figure of 1.13 (instead of 1.12). Should use 5000 iterations instead of 1000, and use the results for 5000 iterations 
Sure, I've implemented this, in the latest git push to this branch. I see that although it is bell curve shape, there are still some modes, which I believe is driven by the variation in characteristics of the sample selected each year, particularly year 5 when a lot of lower BMI individuals with fewer comorbidities start receiving the drug. I've summarised this in [slides 9 & 10](https://docs.google.com/presentation/d/11dfyUIhQ7P22D_-jLaql0tF0Jxof46e2QOa1tbmzvJE/edit?slide=id.g374591b9b5a_0_0#slide=id.g374591b9b5a_0_0).

**SHOULD**:

4. `NUMBER_OF_PEOPLE_PER_YEAR = 1200000` should be removed? We aren’t treated 1.2 million people each year, in this policy?
Yes, that's right. We are not using this constant anywhere in the script. Sorry, I missed deleting it.


5. Amending code to match new NICE categories including adjustment for ethnicity
Not actioning this as per comment above re discussion with PL and project sponsor

6. Should find a way to iterate the Tirzepatide model without copy-pasting large chunks of code to avoid any mistakes (e.g., making changes in one script that don’t get carried over to the second)
This can be implemented directly by calling the utils.R script in the `analysis/policy_analysis/new_policies/policy_38.R` script, just as it is called in the `analysis/policy_analysis/new_policies/sensitivity_testing_policy_38.R`. So then it avoids calling two different sources of the same function. Sorry, I should have addressed this before pushing. Thanks for flagging!

**COULD**:

7. I was able to run analysis/policy_analysis/new_policies/policy_38.R almost the entire script without errors, up until line 663 (writing table_outputs). But I was able to view table_outputs, just not to save it, so that’s not an immediate issue.
I'll pick this up and test next week. I think it could be because a folder is not getting created automatically
